llvm/include/llvm/CodeGen/TargetLowering.h

```diff
@@ -3005,6 +3005,12 @@ class TargetLoweringBase {
     return false;
   }
 
+  /// Return true if sitofp from FromTy to ToTy is cheaper than
+  /// uitofp.
+  virtual bool isSIToFPCheaperThanUIToFP(EVT FromTy, EVT ToTy) const {
+    return false;
```

Comment:
It should be true for X86 target?

Reply:
Yes, alot of other targets too I think. I was planning to followup with patch for targets.

Reply:
Any idea of the list of targets actually? I can just add those as follow up commits, think they will be relatively minor/non-controversial.

Reply:
Okay, added x86 here before we have any actual codegen for nneg

Reply:
I really dislike adding random target hooks like this, especially without reasonable defaults. Just check which of the two is legal/custom like most places?

Reply:
Could you please also add an `LLT` variant like `isZExtFree` above.

Reply:
Can we avoid this hook entirely? X86's combineUIntToFP already folds to SINT_TO_FP/STRICT_SINT_TO_FP if the signbit is zero, just add a check for the nneg flag as well? Other targets can probably do it as easily

Reply:
Think the slight cost of that is we need to adj the impl for each target, but I'm fundementally fine with that.

Reply:
Changes to what your proposed.